src/System.Net.Sockets/src/System/Net/Sockets/UnixDomainSocketEndPoint.Unix.cs

```diff
@@ -70,7 +78,17 @@ internal UnixDomainSocketEndPoint(SocketAddress socketAddress)
                     _encodedPath[i] = socketAddress[s_nativePathOffset + i];
                 }
 
-                _path = s_pathEncoding.GetString(_encodedPath, 0, _encodedPath.Length);
+                // Strip trailing null of pathname socket addresses.
+                int length = _encodedPath.Length;
+                bool isAbstract = _encodedPath[0] == 0;
+                if (!isAbstract)
+                {
+                    if (_encodedPath[length - 1] == 0)
```

Comment:
Is there anything stopping someone from having multiple `\0` characters at the end of the address?

Basically I'm wondering "does this need to be in a loop?".

Reply:
According to the test it doesn't have to. But it doesn't hurt to make it a loop. So I'll change it.